### DISCUSSION TITLE
Getting an executable path or class (?) from a taskListButtonElement

### DISCUSSION THREAD

**Discussion by YellowAfterlife** [ramensoftware/windhawk-mods#788] [Q&A] [answered]
Hey, I've been tweaking the Taskbar Labels mod to bring 7+ Taskbar Tweaker's behaviour with enabling/disabling taskbar labels for individual applications to Windows 11 taskbar.

Right now I'm changing this
https://github.com/ramensoftware/windhawk-mods/blob/3100a61198d0238712e2e303fe366a1d7aa3ac08/mods/taskbar-labels.wh.cpp#L793-L803
to this:
```cpp
    if (secondColumnWidthPixels > 0 && labelControlElement) {
        auto text = labelControlElement.Text();
        auto hide = (text == L"Task Manager"
            || text == L"Vial"
            || text.ends_with(L" - Notepad++")
            || text.ends_with(L" - Windhawk")
            // ... you get the idea
        );
        if (hide) {
            labelControlElement.Visibility(Visibility::Collapsed);
            columnDefinitions.GetAt(1).Width(GridLength({
                .Value = 0,
                .GridUnitType = GridUnitType::Pixel,
            }));
        } else {
            labelControlElement.Visibility(Visibility::Visible);
            columnDefinitions.GetAt(1).Width(GridLength({
                .Value = secondColumnWidthPixels,
                .GridUnitType = GridUnitType::Pixel,
            }));
        }
    } else {
        columnDefinitions.GetAt(1).Width(GridLength({
            .Value = 1,
            .GridUnitType = GridUnitType::Auto,
        }));
    }
```
And this works,  
![image](https://github.com/ramensoftware/windhawk-mods/assets/731492/aa71a13a-75e0-412a-b0c4-bbed28060447)  
but that's a little silly, isn't it? I wouldn't mind a more concrete way of telling which program a button represents.

**Comment by m417z** [comment, marked as answer]
Hi, showing/hiding labels per-program is now possible in version 1.3. See also: https://github.com/ramensoftware/windhawk-mods/issues/203.
